packages/waas-ethers/src/signer.ts

```diff
@@ -0,0 +1,127 @@
+import { BigNumber, ethers } from 'ethers'
+import { CommonAuthArgs, Sequence, networks } from '@0xsequence/waas'
+
+export class SequenceSigner extends ethers.Signer {
+  constructor(
+    private readonly sequence: Sequence,
+    readonly provider?: ethers.providers.BaseProvider
+  ) {
+    super()
+  }
+
+  async getAddress(): Promise<string> {
+    return this.sequence.getAddress()
+  }
+
+  // Ensure that sequence and the provider are on matching networks
+  private async _ensureNetworkMatch(providerRequired: boolean): Promise<void> {
+    if (providerRequired && !this.provider) {
+      throw new Error('Provider is required')
+    }
+    if (this.provider && (await this.provider.getNetwork()).chainId !== networks.toNetworkID(this.sequence.getNetwork())) {
```

Comment:
Sequence Auth supports multiple networks, maybe is it worth to accept any provider as long as the network is among the supported ones by Sequence Auth?